### PR TITLE
Task search 1.0

### PR DESCRIPTION
**Pull request by teastburn**
Follow up from https://github.com/Nextdoor/airflow/pull/30

1) use spinner before load (i should add some screenshots for this)
2) make payload smaller and transform it in JS (671kB to 185kB; transform: 2ms on local chrome for 3500 records)
3) use gzip (185kB to 22.8kB)

### CODE REVIEW

File: airflow/www/templates/admin/master.html [hunk 1]
```diff
@@ -128,64 +128,75 @@
...
-    let key = e.which || e.keyCode;
-    let letterT = 84;
-    if (key === letterT) {
-      initTaskSearchModal();
-    }
+// Get task search data in the background
+$.getJSON("{{ url_for('airflow.all_tasks') }}", function (data) {
+  var typeahead_data = [];
+  for (const [dag, tasks] of Object.entries(data)) {
+    typeahead_data = typeahead_data.concat(tasks.flatMap(task => dag + " " + task));
```
Comment (teastburn) [inline airflow/www/templates/admin/master.html]: this is not right. `typeahead_data` should be a list of objects like 
```json
[
  {"dag_id":"blah1", "task_id":"blah1", "name":"blah1 blah1"},
  {"dag_id":"blah2", "task_id":"blah2", "name":"blah2 blah2"},
]
```

Comment (kavinnd) [reply airflow/www/templates/admin/master.html]: looking up online, concat should work ?

Comment (teastburn) [reply airflow/www/templates/admin/master.html]: yeah it's more that the flatmap returns a string instead of an object

Comment (kavinnd) [reply airflow/www/templates/admin/master.html]: ah right. you need map()

Comment (teastburn) [reply airflow/www/templates/admin/master.html]: i think just this?
```suggestion
    typeahead_data = typeahead_data.concat(tasks.flatMap(task => {"dag_id": dag, "task_id": task, "name": dag + " " + task}));
```

File: airflow/www/views.py [hunk 2]
```diff
@@ -1351,12 +1351,9 @@ def clear(self):
 
     @expose('/all_tasks', methods=['GET'])
     @login_required
+    @wwwutils.gzipped
```
Comment (kavinnd) [inline airflow/www/views.py:1354]: just importing the lib is all you needed?

**Review by kavinnd — APPROVED** [review]

**Review by kavinnd — APPROVED** [review]

File: airflow/www/views.py (hunk 2 above)
Comment (teastburn) [reply airflow/www/views.py:1354]: using a decorator, yeah. airflow team already built it for us!

**Pull request merged**